apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js

```diff
@@ -718,8 +723,21 @@ async function openLinksInParentFrame( calypsoPort ) {
 			}
 		}
 	} );
-	const popoverSlotElem = document.querySelector( 'body' );
-	popoverSlotElem && popoverSlotObserver.observe( popoverSlotElem, { childList: true } );
+	const popoverSlotObserverObserver = new window.MutationObserver( ( mutations ) => {
+		for ( const record of mutations ) {
+			for ( const node of record.addedNodes ) {
+				if ( ! node ) {
+					continue;
+				}
+				if ( node?.classList?.contains( 'components-popover__fallback-container' ) ) {
+					popoverSlotObserver.observe( node, { childList: true } );
+				}
+			}
+		}
+	} );
+	const htmlBody = document.querySelector( 'body' );
+	htmlBody && popoverSlotObserver.observe( htmlBody, { childList: true } );
+	htmlBody && popoverSlotObserverObserver.observe( htmlBody, { childList: true } );
```

Comment:
Review note: Both are required, the first should only be getting called once due to `components-popover__fallback-container` only being called once.

Skipping the first means it never observes the initial link inserts, you need to close and reopen to see updated links.

Skipping the second means it never observes the secondary link inserts, you don't see the updated link on reopen.

Reply:
I think we have lots of assumptions here such as 
* The popover component would be inserted into an element under the body
* The `.components-popover__fallback-container` element might be the element to be inserted

How about just copying [these lines](https://github.com/WordPress/gutenberg/blob/8ba6b11fba4e12bec3ad6174886fbcc8a97a0094/packages/components/src/popover/index.tsx#L97-L109) and calling `getPopoverFallbackContainer` to get the container and observe its child list to make it simple?

Reply:
I'm trying this idea now and will take over the PR.

Reply:
Thanks for the suggestion, this simplifies things a bit!